src/main/kotlin/cacophony/automata/minimalization/DFAUtils.kt

```diff
@@ -76,4 +79,9 @@ fun <DFAState> DFA<DFAState>.withStates(retain: Set<DFAState>): DFA<DFAState> {
     )
 }
 
-infix fun <DFAState> DFAState.via(label: Char): Pair<DFAState, Char> = Pair(this, label)
+infix fun <DFAState, Atom> DFAState.via(label: Atom): Pair<DFAState, Atom> = Pair(this, label)
+
+// Some utility functions, to not write whole pipeline each time
+fun buildDFAFromRegex(regex: AlgebraicRegex) = determinize(buildNFAFromRegex(regex)).minimalize()
+
+fun buildDFAFromRegex(regex: String) = buildDFAFromRegex(AlgebraicRegex.fromString(regex))
```

Comment:
I feel like this should not be in `minimalization` package 😄

Reply:
Yeah, placed it there so that it can be moved along with 'via' during refctor

Reply:
if you want to leave something for the refactor, please write a comment in https://github.com/mhorod/cacophony/issues/58